### PR TITLE
Remove abandoned cart functionality

### PR DESCRIPTION
**Pull request by jacobwinch**
### Why do we need this?
The code which uses this functionality is being removed [here](https://github.com/guardian/membership-frontend/pull/1809) and we want to tidy up any AWS resources related to this feature.

I'll follow up with a second PR to remove any Cloudformation stuff related to this functionality.

### The changes
* Remove (soon to be) unused code related to abandoned cart functionality

### trello card/screenshot/json/related PRs etc
I need to do this so that I can complete:
https://trello.com/c/wneCUDJH/330-delete-the-membership-behaviour-dynamo-table-e1

### CODE REVIEW

File: membership-attribute-service/app/components/TouchpointComponents.scala [hunk 1]
```diff
@@ -34,12 +33,6 @@ class TouchpointComponents(stage: String)(implicit  system: ActorSystem, executi
...
 
-  lazy val identityConfig = new IdentityConfig {
-    override def token: String = environmentConf.getString("identity.marketingToken")
-    override def url: String = environmentConf.getString("identity.apiUrl")
-  }
-
-  lazy val identityService = new IdentityService(identityConfig, RequestRunners.futureRunner)
   lazy val digitalPackConf = environmentConf.getConfig(s"zuora.ratePlanIds.digitalpack")
   lazy val paperCatalogConf = environmentConf.getConfig(s"zuora.productIds.subscriptions")
   lazy val membershipConf = environmentConf.getConfig(s"zuora.ratePlanIds.membership")
```
Comment (lmath) [inline membership-attribute-service/app/components/TouchpointComponents.scala]: Can dynamoBehaviourTable be removed here?

Comment (jacobwinch) [reply membership-attribute-service/app/components/TouchpointComponents.scala]: Yep, good catch - thanks

File: membership-attribute-service/app/controllers/BehaviourController.scala [hunk 2]
```diff
@@ -1,142 +0,0 @@
...
-  }
-
-  def sendCartReminderEmail = BackendFromCookieAction.async { implicit request =>
-    val receivedBehaviour = behaviourFromBody(request.body.asJson)
-    for {
-      paidTier <- request.touchpoint.attrService.get(receivedBehaviour.userId).map(_.exists(_.isPaidTier))
-      user <- request.touchpoint.identityService.user(IdentityId(receivedBehaviour.userId))
-      emailAddress = (user \ "user" \ "primaryEmailAddress").asOpt[String]
-      displayName = (user \ "user" \ "publicFields" \ "displayName").asOpt[String]
-      gnmMarketingPrefs = true // TODO - needs an Identity API PR to send statusFields.receiveGnmMarketing in above user <- ... call
```
Comment (paulbrown1982) [inline membership-attribute-service/app/controllers/BehaviourController.scala]: 😨

File: membership-attribute-service/app/services/IdentityService.scala [hunk 3]
```diff
@@ -1,43 +0,0 @@
...
-import play.api.libs.json.{JsValue, Json, __}
-import java.net.URLEncoder.encode
-
-import IdentityService._
-import com.gu.okhttp.RequestRunners
-import okhttp3.{MediaType, Request, RequestBody, ResponseBody}
-
-import scala.concurrent.{ExecutionContext, Future}
-
-object IdentityService {
```
Comment (paulbrown1982) [inline membership-attribute-service/app/services/IdentityService.scala]: Might we need this later in the week to do the saves impression tracking?

Comment (jacobwinch) [reply membership-attribute-service/app/services/IdentityService.scala]: I had assumed that the saves tracking would be done using GA? Perhaps we can discuss offline if I've misunderstood?

Comment (paulbrown1982) [reply membership-attribute-service/app/services/IdentityService.scala]: I was thinking more about automating what @lmath has been doing via ELK analysis. 

Also FYI, we must no longer put Identity ID into GA.

Comment (jacobwinch) [reply membership-attribute-service/app/services/IdentityService.scala]: OK, I guess we can discuss further once we get to that card. For now I think I'll just remove this code and we can re-instate it if we need it later...

**Pull request merged**